exploration-module/src/main/java/io/edpn/backend/exploration/application/service/ReceiveJournalDockedService.java

```diff
@@ -0,0 +1,119 @@
+        executorService.submit(ConcurrencyUtil.errorHandlingWrapper(
+                () -> systemEliteIdResponseSender.sendResponsesForSystem(station.system().name()),
+                exception -> log.error("Error while sending systemEliteIdResponse for system: {}", station.system().name(), exception)));
+        executorService.submit(ConcurrencyUtil.errorHandlingWrapper(
+                () -> stationMaxLandingPadSizeResponseSender.sendResponsesForStation(station.system().name(), station.name()),
+                exception -> log.error("Error while sending stationMaxLandingPadSizeResponse for station: {}", station.name(), exception)));
+    }
+
+    private Map<LandingPadSize, Integer> landingPadSizeFromPayload(DockedMessage.V1.Payload payload) {
+        return Map.of(
+                LandingPadSize.LARGE, payload.landingPads().large(),
+                LandingPadSize.MEDIUM, payload.landingPads().medium(),
+                LandingPadSize.SMALL, payload.landingPads().small()
+        );
+    }
+
+    private Map<String, Double> stationEconomiesFromPayload(DockedMessage.V1.Payload payload) {
+        return payload.stationEconomies().stream().
+                collect(Collectors.toMap(
+                        DockedMessage.V1.Payload.Economy::name,
+                        DockedMessage.V1.Payload.Economy::proportion,
+                        (first, second) -> first
+                ));
+    }
+
+    private Station saveOrUpdateStationFromPayload(DockedMessage.V1.Payload payload, LocalDateTime messageTimeStamp) {
+        return saveOrUpdateStationPort.saveOrUpdate(
+                new Station(
+                        idGenerator.generateId(),
+                        payload.marketId(),
+                        payload.stationName(),
+                        payload.stationType(),
+                        payload.distanceFromStar(),
+                        saveOrUpdateSystemFromPayload(payload),
+                        landingPadSizeFromPayload(payload),
+                        stationEconomiesFromPayload(payload),
+                        payload.stationServices(),
+                        payload.stationEconomy(),
+                        payload.stationGovernment(),
+                        payload.odyssey(),
+                        payload.horizons(),
+                        messageTimeStamp));
+    }
+
+    private System saveOrUpdateSystemFromPayload(DockedMessage.V1.Payload payload) {
+        return saveOrUpdateSystemPort.saveOrUpdate(
+                new System(
+                        idGenerator.generateId(),
+                        payload.systemAddress(),
+                        payload.starSystem(),
+                        null,
+                        starPositionToCoordinate(payload.starPosition())));
+    }
+
+    private Coordinate starPositionToCoordinate(Double[] starPosition) {
```

Comment:
I think this kind of questions the choice of representing position as an array in the first place?